react-client/src/components/Accounts/Accounts.tsx

```diff
@@ -129,25 +132,108 @@ const Accounts = () => {
-        <Divider my='sm' label={i18n.get['DisplayName']} />
+      <form onSubmit={userProfileForm.onSubmit(handleUserProfileSubmit)}>
+        <Divider my='sm' label={i18n.get['Profile']} />
         <TextInput
           label={i18n.get['DisplayName']}
           name='displayName'
-          {...userDisplayNameForm.getInputProps('displayName')}
+          {...userProfileForm.getInputProps('displayName')}
         />
-        <Group position='right' mt='md'>
-          <Button type='submit'>
-            {i18n.get['Apply']}
-          </Button>
-        </Group>
+        <Input.Wrapper label={i18n.get['Avatar']}>
+          <Dropzone
+            name='avatar'
+            maxSize={2 * 1024 **2}
+            accept={IMAGE_MIME_TYPE}
+            multiple={false}
+            styles={{ inner: { pointerEvents: 'all' } }}
+            onDrop={(files: FileWithPath[]) => {
+              const reader = new FileReader();
+              reader.onload = () => {
+                const result = reader.result as string;
+                if (result) {
+                  setAvatar(result);
+                  userProfileForm.setFieldValue('avatar', result);
+                }
+              }
+              files[0] && reader.readAsDataURL(files[0]);
+            }}
+          >
+            <Group justify='center'>
+              <Dropzone.Accept>
+                <PhotoUp />
+              </Dropzone.Accept>
+              <Dropzone.Reject>
+                <PhotoX />
+              </Dropzone.Reject>
+              <Dropzone.Idle>
+                <div onClick={e => {e.stopPropagation()}}>
+                  <HoverCard disabled={avatar === ''}>
+                    <HoverCard.Target>
+                      <Avatar radius='xl' size='lg' src={avatar !== '' ? avatar : null}>
+                        {avatar === '' && <User size={24} />}
+                      </Avatar>
+                    </HoverCard.Target>
+                    <HoverCard.Dropdown>
+                      <Button
+                        onClick={ () => {
+                          const newavatar = (user.avatar && avatar !== user.avatar) ? user.avatar : '';
+                          setAvatar(newavatar);
+                          userProfileForm.setFieldValue('avatar', newavatar);
+                        }}
+                      >
+                        Delete
+                      </Button>
+                    </HoverCard.Dropdown>
+                  </HoverCard>
+                </div>
+              </Dropzone.Idle>
+              <div>
+                <Text inline>
+                  Drag image here or click to select file
+                </Text>
+                <Text size='sm' c='dimmed' inline mt={7}>
+                  File should not exceed 2mb
+                </Text>
+              </div>
+            </Group>
+          </Dropzone>
+        </Input.Wrapper>
+        {/* removed until root user choice is implemented
```

Comment:
What is the plan for this? Do we need an issue to prevent it from getting forgotten?